sos/plugins/kubernetes.py

```diff
@@ -15,40 +15,130 @@
+            kube_bin = "kubectl "
+            if path.exists('/etc/origin/master/admin.kubeconfig'):
+                kube_bin += "--config=/etc/origin/master/admin.kubeconfig"
+
+            kube_get = "get -o json "
+            if self.get_option('all'):
+                kube_get += "--all-namespaces=true"
+
+            for cmd in ['version', 'config view']:
+                self.add_cmd_output("{} {}".format(kube_bin, cmd))
+
+            resources = [
+                'events', 'limitrange', 'pods', 'pvc', 'rc',
+                'resourcequota', 'services'
+            ]
+
+            for res in resources:
+                self.add_cmd_output(
+                    "{} {} {}".format(
+                        kube_bin,
+                        kube_get,
+                        res
+                    )
+                )
+            resources.remove('events')
+            # nodes and pvs are not namespaced, must pull separately
+            self.add_cmd_output([
+                "{} get -o json nodes".format(kube_bin),
+                "{} get -o json pv".format(kube_bin)
+                ]
+            )
+
+            if self.get_option('describe'):
+                resources = resources + ['nodes', 'namespaces']
+                for res in resources:
+                    r = self.get_command_output(
+                        "{} get {}".format(
+                            kube_bin,
+                            res
+                        )
+                    )
+                    if r['status'] == 0:
+                        k_list = [k.split()[0] for k in
+                                  r['output'].splitlines()[1:]
+                                  ]
+                        for k in k_list:
+                            self.add_cmd_output(
+                                "{} describe {} {}".format(
+                                    kube_bin,
+                                    res,
+                                    k
+                                )
+                            )
+
+            if self.get_option('podlogs'):
+                r = self.get_command_output(
+                    "{} get pods".format(kube_bin)
+                )
+                if r['status'] == 0:
+                    pods = [p.split()[0] for p in
+                            r['output'].splitlines()[1:]
+                            ]
+                    for pod in pods:
+                        self.add_cmd_output(
+                            "{} logs {}".format(kube_bin, pod)
+                        )
+
+    def postproc(self):
+        # First, clear sensitive data from the json output collected.
+        # This will mask values when the "name" looks susceptible of
+        # values worth obfuscating, i.e. if the name contains strings
+        # like "pass", "pwd", "key" or "token"
+        env_regexp = r'(?P<var>{\s*"name":\s*[^,]*' \
+                     r'(pass|pwd|key|token|cred)[^,]*,\s*"value":)[^}]*'
```

Comment:
Isn't this a variation on `ovirt`'s `sensitive_keys` mechanism? We'd like to unify them at some stage and make the pattern available as a standard `Plugin` interface - the more they have in common the better to enable this work.